core/src/main/scala/akka/persistence/r2dbc/state/scaladsl/DurableStateDao.scala

```diff
@@ -683,17 +671,34 @@ private[r2dbc] class DurableStateDao(settings: R2dbcSettings, connectionFactory:
   }
 
   def persistenceIds(afterId: Option[String], limit: Long): Source[String, NotUsed] = {
+    if (settings.durableStateTableByEntityTypeWithSchema.isEmpty)
+      persistenceIds(afterId, limit, settings.durableStateTableWithSchema)
+    else {
+      val customTables = settings.durableStateTableByEntityTypeWithSchema.toVector.sortBy(_._1).map(_._2)
+      persistenceIds(afterId, limit, settings.durableStateTableWithSchema)
+        .concatAllLazy(customTables.map(table => persistenceIds(afterId, limit, table)): _*)
```

Comment:
Note that this materializes eagerly, `concatAllLazy` is just lazy in the sense that it doesn't pull until there is demand (vs concat which pulls one element immediately). Not sure if that is a problem here, can something timeout, is the query triggered on materialization or demand? 

Can be made lazy in the sense you expect by wrapping custom table queries with `Source.lazy`

Reply:
Gotcha, I'll take a look

Reply:
I rewrote this in 537dd8a because the sort order was pretty useless